File: tdd_practice/test_demo/src/main/java/com/example/test_demo/user/infrastructure/UserRepositoryImpl.java
package com.example.test_demo.user.infrastructure;

import com.example.test_demo.user.domain.UserStatus;
import com.example.test_demo.user.service.port.UserRepository;
import lombok.RequiredArgsConstructor;
import org.springframework.stereotype.Repository;

import java.util.Optional;

@Repository
@RequiredArgsConstructor
public class UserRepositoryImpl implements UserRepository {

    private final UserJpaRepository userJpaRepository;

    @Override
    public Optional<UserEntity> findByEmailAndStatus(String email, UserStatus userStatus) {
        return userJpaRepository.findByEmailAndStatus(email, userStatus);
    }

    @Override
    public Optional<UserEntity> findByIdAndStatus(long id, UserStatus userStatus) {
        return userJpaRepository.findByIdAndStatus(id, userStatus);
    }

    @Override
    public UserEntity save(UserEntity userEntity) {
        return userJpaRepository.save(userEntity);
    }

    @Override
    public Optional<UserEntity> findById(long id) {
        return userJpaRepository.findById(id);
    }
}

File: social-login/gateway/db/src/main/kotlin/org/example/sociallogin/gateway/db/user/UserRepositoryImpl.kt
package org.example.sociallogin.gateway.db.user

import org.example.sociallogin.domain.user.User
import org.example.sociallogin.domain.user.UserRepository
import org.springframework.stereotype.Repository
import reactor.core.publisher.Mono

@Repository
class UserRepositoryImpl(
    private val userR2dbcRepository: UserR2dbcRepository,
): UserRepository {

    override fun save(user: User): Mono<User> {
        val userEntity = UserEntity(
            name = user.name,
            email = user.email,
            socialId = user.socialId,
            provider = user.provider,
        )
        return userR2dbcRepository.save(userEntity)
            .map { it.toDomain() }
    }
}